sys/net/netstats/netstats_neighbor.c

```diff
@@ -0,0 +1,410 @@
+
+    return res;
+}
+
+__attribute__((unused))
+static uint32_t _ewma(bool fresh, uint32_t old_val, uint32_t new_val)
+{
+    uint8_t ewma_alpha;
+
+    if (old_val == 0) {
+        return new_val;
+    }
+
+    /* If the stats are not fresh, use a larger alpha to average aggressive */
+    if (fresh) {
+        ewma_alpha = NETSTATS_NB_EWMA_ALPHA;
+    } else {
+        ewma_alpha = NETSTATS_NB_EWMA_ALPHA_RAMP;
+    }
+
+    /* Exponential weighted moving average */
+    return (old_val * (NETSTATS_NB_EWMA_SCALE - ewma_alpha)
+         +  new_val * ewma_alpha) / NETSTATS_NB_EWMA_SCALE;
+}
+
+static void netstats_nb_update_etx(netstats_nb_t *stats, netstats_nb_result_t result,
+                                   uint8_t transmissions, bool fresh)
+{
+    /* don't do anything if driver does not report ETX */
+    if (transmissions == 0) {
+        return;
+    }
+
+    if (result != NETSTATS_NB_SUCCESS) {
+        transmissions = NETSTATS_NB_ETX_NOACK_PENALTY;
+    }
+
+#ifdef MODULE_NETSTATS_NEIGHBOR_ETX
+    stats->etx = _ewma(fresh, stats->etx, transmissions * NETSTATS_NB_ETX_DIVISOR);
+#else
+    (void)stats;
+    (void)result;
+    (void)transmissions;
+    (void)fresh;
+#endif
+}
+
+static void netstats_nb_update_time(netstats_nb_t *stats, netstats_nb_result_t result,
+                                    uint32_t duration, bool fresh)
+{
+    /* TX time already got a penalty due to retransmissions */
+    if (result != NETSTATS_NB_SUCCESS) {
+        duration *= 2;
+    }
+
+#if MODULE_NETSTATS_NEIGHBOR_TX_TIME
+    stats->time_tx_avg = _ewma(fresh, stats->time_tx_avg, duration);
+#else
+    (void)stats;
+    (void)result;
+    (void)duration;
+    (void)fresh;
+#endif
+}
+
+static void netstats_nb_update_rssi(netstats_nb_t *stats, uint8_t rssi, bool fresh)
+{
+#ifdef MODULE_NETSTATS_NEIGHBOR_RSSI
+    stats->rssi = _ewma(fresh, stats->rssi, rssi);
+#else
+    (void)stats;
+    (void)rssi;
+    (void)fresh;
+#endif
+}
+
+static void netstats_nb_update_lqi(netstats_nb_t *stats, uint8_t lqi, bool fresh)
+{
+#ifdef MODULE_NETSTATS_NEIGHBOR_LQI
+    stats->lqi = _ewma(fresh, stats->lqi, lqi);
+#else
+    (void)stats;
+    (void)lqi;
+    (void)fresh;
+#endif
+}
+
+static void netstats_nb_incr_count_tx(netstats_nb_t *stats, netstats_nb_result_t result)
+{
+#ifdef MODULE_NETSTATS_NEIGHBOR_COUNT
+    stats->tx_count++;
+
+    /* gracefully handle overflow */
+    if (stats->tx_count == 0) {
+        stats->tx_count = ~stats->tx_count;
+        stats->tx_count = stats->tx_count >> 4;
+        stats->tx_fail  = stats->tx_fail  >> 4;
+    }
+
+    if (result != NETSTATS_NB_SUCCESS) {
+        stats->tx_fail++;
+    }
+#else
+    (void)stats;
+    (void)result;
+#endif
+}
+
+static void netstats_nb_incr_count_rx(netstats_nb_t *stats)
+{
+#ifdef MODULE_NETSTATS_NEIGHBOR_COUNT
+    stats->rx_count++;
+#else
+    (void)stats;
+#endif
+}
+
+netstats_nb_t *netstats_nb_update_tx(netif_t *dev, netstats_nb_result_t result, uint8_t transmissions)
```

Comment:
Line longer than 100 characters says `vera++`. There is an easy fix :-).
```suggestion
netstats_nb_t *netstats_nb_update_tx(netif_t *dev, netstats_nb_result_t result, 
                                     uint8_t transmissions)
```